apps/dav/lib/CardDAV/SyncService.php

```diff
@@ -121,7 +125,8 @@ protected function requestSyncReport(string $url, string $userName, string $addr
 			'auth' => [$userName, $sharedSecret],
 			'base_uri' => $url,
 			'body' => $this->buildSyncCollectionRequestBody($syncToken),
-			'headers' => ['Content-Type' => 'application/xml']
+			'headers' => ['Content-Type' => 'application/xml'],
+			'timeout' => $this->config->getSystemValueInt('carddav_sync_request_timeout', 30)
```

Comment:
```suggestion
			'timeout' => $this->config->getSystemValueInt('carddav_sync_request_timeout', IClient::DEFAULT_REQUEST_TIMEOUT)
```

(add `use OCP\Http\Client\IClient;` as well)

Reply:
We were thinking of backporting this. To save time and skip another review round, we can just do the constant migration in a follow-up.